lib/routes.js

```diff
@@ -21,58 +55,64 @@ var setup = function (app, router) {
-    req.query.debug = (req.query.hasOwnProperty('debug')) ? req.query.debug : false
-
-
-    // Reading
-    modbus.readRegister(
-      req.params.ip,
-      req.params.port,
-      req.params.unitId,
-      req.params.regType,
-      req.params.startAddress,
-      req.params.quantity,
-      req.query.conversion,
-      req.query.endianness
-    ).then((reading) => {
-      if (isNaN(reading) || reading == null) {
-        res.status(400).json({ "params": req.params, "reading": String(reading), "error": "NaN" })
-      } else {
-        // Scaling
-        req.query.multiplier = (req.query.hasOwnProperty('multiplier')) ? req.query.multiplier : 1
-        req.query.offset = (req.query.hasOwnProperty('offset')) ? req.query.offset : 0
-        const transformedReading = transformReading(reading, req.query.offset, req.query.multiplier)
-
-        // Respond
-        if (req.query.debug) {
-          res.status(200).json({ "params": req.params, "query": req.query, "reading": reading, "transformedReading": transformedReading })
+    // Add item to queue with callback
+    q.push({ name: 'Modbus Reading' }, function (err) {
+      if (err) {
+        // Handle error
+        console.error(err);
+        return res.status(500).json({ error: 'Internal Server Error' });
+      }
+
+      let reading = null
+
+      // Conversion Values
+      req.query.conversion = (req.query.hasOwnProperty('conversion')) ? req.query.conversion : "interger"
+      req.query.endianness = (req.query.hasOwnProperty('endianness')) ? req.query.endianness : "big"
+      req.query.debug = (req.query.hasOwnProperty('debug')) ? req.query.debug : false
+
+      // Reading
+      modbus.readRegister(
+        req.params.ip,
+        req.params.port,
+        req.params.unitId,
+        req.params.regType,
+        req.params.startAddress,
+        req.params.quantity,
+        req.query.conversion,
+        req.query.endianness
+      ).then((reading) => {
+        if (isNaN(reading) || reading == null) {
+          return res.status(400).json({ "params": req.params, "reading": String(reading), "error": "NaN" })
         } else {
-          res.status(200).json({ "reading": reading, "transformedReading": transformedReading })
+          // Scaling
+          req.query.multiplier = (req.query.hasOwnProperty('multiplier')) ? req.query.multiplier : 1
+          req.query.offset = (req.query.hasOwnProperty('offset')) ? req.query.offset : 0
+          const transformedReading = transformReading(reading, req.query.offset, req.query.multiplier)
+
+          // Respond
+          if (req.query.debug) {
+            return res.status(200).json({ "params": req.params, "query": req.query, "reading": reading, "transformedReading": transformedReading })
+          } else {
+            return res.status(200).json({ "reading": reading, "transformedReading": transformedReading })
+          }
+
         }
-      }
-    })
+      }).catch((error) => {
```

Comment:
When you use `await`, exception handling goes back to the standard `try {...} catch (error) {...} finally {...}` construction. This wants to go round the whole lot, including the call to `modbus.readRegister`.